File: VB/XPCustomBindingExample/XPO/BasePersistentClass.vb
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata

Namespace XPCustomBindingExample.DAL
	<NonPersistent>
	Public Class BasePersistentClass
		Inherits XPLiteObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Sub New(ByVal session As Session, ByVal classInfo As XPClassInfo)
			MyBase.New(session, classInfo)
		End Sub
	End Class
End Namespace

File: CS/XPCustomBindingExample/XPO/BasePersistentClass.cs
using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;

namespace XPCustomBindingExample.DAL {
    [NonPersistent]
    public class BasePersistentClass : XPLiteObject {
        public BasePersistentClass(Session session) : base(session) { }
        public BasePersistentClass(Session session, XPClassInfo classInfo) : base(session, classInfo) { }
    }
}